versions/v1/index.ts

```diff
@@ -1,195 +1,235 @@
+        helpersFile,
+      };
+
+      const server = await createServer(config, typeDefs, resolvers, helpers);
+      await server.listen();
+      await server.stop();
+    } catch (e) {
+      console.error("❌ Error validating", e);
+      process.exit(1);
+    }
+
+    console.log("✅ Validated successfully");
+  },
+  local: async (args) => {
+    validateFiles();
+    await validateExports();
+    const deployKey = args.deployKey || getDeployKey(MD_PATH);
+    const cliData = await getSatsumaMetadata(
+      args.subgraphName,
+      args.versionName,
+      deployKey
+    );
+    if (!cliData) {
+      return;
+    }
+
+    const tables: Database["tables"] = {};
+
+    // Non-partitioned tables with block range column
+    for (const table of cliData.nonPartitionedTablesBR) {
+      tables[`${table}__all`] = {
+        actualName: table,
+      };
+      tables[table] = {
+        actualName: table,
+        whereClause: `block_range @> 2147483647`,
+      };
+    }
+
+    // Non-partitioned tables without block range column. Usually immutable.
+    for (const table of cliData.nonPartitionedTables) {
+      if (tables[table]) {
+        // Must have been added already with block range
+        continue;
+      }
+
+      // Tables with no block range column we just provide the whole thing as if it's active
+      tables[table] = {
+        actualName: table,
+      };
+    }
+
+    // Partitioned tables always have block range column. That's what the partition is on
+    for (const table of cliData.partitionedTables) {
+      tables[table] = {
+        actualName: `${table}_active`,
+      };
+      tables[`${table}__all`] = {
+        actualName: `${table}_inactive`,
+      };
+    }
+
+    // Convert into the format that the codegen expects
+    const databases: CreateServerConfig["databases"] = [
+      {
+        uri: cliData.dbUri,
+        type: "pg" as const,
+        name: "knex",
+        search_path: cliData.entitySchema,
+        tables: {},
+      },
+    ];
+    const graphql = [
+      {
+        uri: cliData.queryHost,
+      },
+    ];
+
+    try {
+      const {
+        typeDefs,
+        resolvers,
+        helpers,
+        resolverFile,
+        typeDefsFile,
+        helpersFile,
+      } = await loadCustomerCode();
+      const config: CreateServerConfig = {
+        databases,
+        graphql,
+        resolverFile,
+        typeDefsFile,
+        helpersFile,
+      };
+
+      const server = await createServer(config, typeDefs, resolvers, helpers);
+      // eslint-disable-next-line no-async-promise-executor
+      return new Promise(async () => {
+        const { url } = await server.listen();
+        console.log(`🍊Satsuma server listening at ${url}`);
+
+        const shutdownServer = () => {
+          console.log("Shutting down server...");
+          server
+            .stop()
+            .then(() => {
+              process.exit(0);
+            })
+            .catch(() => process.exit(1));
```

Comment:
I added a catch to exit the process if there are rejections from stopping the server.

Seems like we dont care to resolve or reject this promise? It's a bit suss of a style.